src/app/screens/Accounts/index.tsx

```diff
@@ -1,22 +1,44 @@
 import {
   EllipsisIcon,
   PlusIcon,
-  WalletIcon,
+  WalletIcon
 } from "@bitcoin-design/bitcoin-icons-react/filled";
+import { CrossIcon } from "@bitcoin-design/bitcoin-icons-react/outline";
+import type { FormEvent } from "react";
+import { useState } from "react";
+import Modal from "react-modal";
 import { useNavigate } from "react-router-dom";
-
-import api from "../../common/lib/api";
-import Button from "../components/Button";
-import Container from "../components/Container";
-import Menu from "../components/Menu";
-import { useAuth } from "../context/AuthContext";
-import { useAccounts } from "../context/AccountsContext";
+import api from "../../../common/lib/api";
+import utils from "../../../common/lib/utils";
+import Button from "../../components/Button";
+import Container from "../../components/Container";
+import TextField from "../../components/Form/TextField";
+import Menu from "../../components/Menu";
+import { useAccounts } from "../../context/AccountsContext";
+import { useAuth } from "../../context/AuthContext";
 
 function AccountsScreen() {
   const auth = useAuth();
   const { accounts, getAccounts } = useAccounts();
   const navigate = useNavigate();
 
+  const [modalIsOpen, setIsOpen] = useState(false);
+  const [newAccountName, setNewAccountName] = useState("");
+
+  function closeModal() {
+    setIsOpen(false);
+  }
+
+  async function updateAccountName(accountId: string) {
```

Comment:
it seems the accountId is not passed properly here. 
In my test it was always the same.

I have multiple accounts but the ID here was always the ID of the last account in the list

Reply:
Issue is that currently all modals wil be rendered, butyou only see the one on top, which is the last one.  
Will fix.